libmscore/barline.cpp

```diff
@@ -43,13 +43,19 @@ static void undoChangeBarLineType(BarLine* bl, BarLineType barType)
             case BarLineType::BROKEN:
             case BarLineType::DOTTED: {
                   Segment* segment        = bl->segment();
+                  for (ScoreElement* el : bl->linkList()) {
+                        if (el->score()->isMaster())
+                              segment = static_cast<BarLine*>(el)->segment();
```

Comment:
Do you really need `static_cast` here? `segment` is a `Segment*`, `segment()` returns `Segment*` as well.

Reply:
I'm not sure what you meant. ScoreElement doesn't have a segment property. But I replaced the static cast with the toBarLine method and added a isBarLine check before converting.